### ISSUE TITLE
Temporary file/directory support

### ISSUE THREAD

**Issue by heinezen** [SFTtech/openage#1540]
**Required Skills:** Python or C++

**Difficulty**: Easy

It would be nice if we could create temporary files/directories in our current file system abstraction API. We currently only use temporary structures in the converter, but they could be useful in other places, e.g. for unzipping archives or on-disk caching.

Creation of temporary files could be implemented in our [Python file API](https://github.com/SFTtech/openage/tree/master/openage/util/fslike) or our [C++ file API](https://github.com/SFTtech/openage/tree/master/libopenage/util/fslike) or in both. The Python-side implementation is more preferrable at the moment, since the converter is written Python and temporary files already have an active use case there.

The easiest way to implement support probably is to add a method that calls the underlying system libraries for temporary file/directory creation:

- Python: `tempfile` module
- C++: `std::filesystem` and `std::tmpfile`

The retrieved path is then wrapped in our `Path`/`Directory` classes and returned by the method. 

Tasks:
- [ ] Create temp file/dir and wrap them in openage file system API
- [ ] Delete temp file/dir explicitely when object is destroyed
- [ ] Use new implementation in code (currently only in converter)
- [ ] (optional) Add method for checking if a file/dir is temporary

**Further Reading**

* [openage Python file API](https://github.com/SFTtech/openage/tree/master/openage/util/fslike) 
* [openage C++ file API](https://github.com/SFTtech/openage/tree/master/libopenage/util/fslike)
* [Python `tempfile` module](https://docs.python.org/3/library/tempfile.html)
* [C++ std::filesystem::temp_directory_path](https://en.cppreference.com/w/cpp/filesystem/temp_directory_path)
* [C++ std::tmpfile](https://en.cppreference.com/w/cpp/io/c/tmpfile)

**Comment by Ashhar-24** [comment]
Hey @heinezen , I'm interested in working with this issue.
I'm new to open source but I've beginner level experience in working with `Python` and `C++` both.
Can you please assign me this issue.

And also can you please elaborate more about the tasks that exactly what I've to do.
Thanks for your cooperation :)

**Comment by heinezen** [comment]
@Ashhar-24 Hey, I assigned you to to the issue nw :)

Most of what we want from the task should be in the issue description already. The general goal is to make temporary file creation/access more easier than it is right now.

For a start, I think you should try out these steps:

1. [Compile and run](https://github.com/SFTtech/openage?tab=readme-ov-file#quickstart) the project one if you haven't already. Development works much better when you can test your own code :D
2. Look at the current usage of `tempfile` in our code, e.g.
https://github.com/SFTtech/openage/blob/821a6d1a46e2ff7c5f66eeb34d9ece983e9c03b8/openage/convert/tool/subtool/acquire_sourcedir.py#L253-L295
3. openage has its own [path interface](https://github.com/SFTtech/openage/blob/master/openage/util/fslike/path.py) that acts as a wrapper around system paths. It should be extended to get tempfile support. I think the easiest way to implement it in there is to add a staticmethod that first creates a temporary file using `tempfile`, then wraps it in our `Path` object and returns the reulting `Path`.

I think this should give you a good idea how to start. If you get stuck, you can ask more questions in the Matrix chat.

**Comment by viswes31** [comment]
Hey @heinezen , I'm interested in working with this issue.
I'm new to open source but I've 10+ experience Python and C++ both.
I see some of the tasks in this issue are still open. 
Can I work on this issue ?

Thank you :)

**Comment by TheJJ** [comment]
go for it!

**Comment by heinezen** [comment]
@viswes31 Everything from the tasklist that's not ticked is still up for grabs ^^

**Comment by alamt22** [comment]
Hi @heinezen @TheJJ , I understand this issue is a year old, so I was wondering if it's all good for me to work on this issue.
I can start with the second task.

**Comment by heinezen** [comment]
@alamt22 Yeah, there are a few tasks in the list that are still open, so you can try to solve them :)

**Comment by TheJJ** [comment]
yea, please submit a pull request or ask questions (ideally in the chat) :)
